smart-home-application/Device.cpp

```diff
@@ -0,0 +1,63 @@
+#include "Device.h"
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+namespace smart_home_app {
+Device::Device(std::string device_name, bool status) : device_name_(device_name), status_(status) {};
+
+Device::~Device() {};
+
+Device::Device(const Device& other) : device_name_(other.device_name_), status_(other.status_) {};
+
+Device::Device(Device&& other) = default;
+
+Device& Device::operator=(const Device& rhs) {
+    Device temp(rhs.device_name_, rhs.status_);
+    for (Sensor s : rhs.sensors_) {
+        this->sensors_.push_back(s);
+    }
+    std::swap(device_name_, temp.device_name_);
+    std::swap(status_, temp.status_);
+    std::swap(sensors_, temp.sensors_);
+    return *this;
+};
+
+Device& Device::operator=(Device&& rhs) = default;
+
+void Device::SetDeviceName(std::string device_name){
+    this->device_name_ = device_name;
+};
+
+std::string Device::GetDeviceName(){
+    return this->device_name_;
+};
+
+void Device::SetStatus(bool status) {
+    this->status_ = status;
+};
+
+bool Device::GetStatus() {
+    return this->status_;
+};
+
+void Device::AddSensor(Sensor sensor) {
```

Comment:
although is correct, you should pass this as a const reference to improve performance. read a bit more about const and references

this comment applies to other methods as well